include/picongpu/plugins/PluginController.hpp

```diff
@@ -24,177 +24,61 @@
 
 #include "picongpu/simulation_defines.hpp"
 
+#include "picongpu/particles/traits/SpeciesEligibleForSolver.hpp"
 #include "picongpu/plugins/BinEnergyParticles.hpp"
+#include "picongpu/plugins/ChargeConservation.hpp"
+#include "picongpu/plugins/Checkpoint.hpp"
 #include "picongpu/plugins/CountParticles.hpp"
 #include "picongpu/plugins/Emittance.hpp"
 #include "picongpu/plugins/EnergyFields.hpp"
 #include "picongpu/plugins/EnergyParticles.hpp"
+#include "picongpu/plugins/ILightweightPlugin.hpp"
+#include "picongpu/plugins/ISimulationPlugin.hpp"
+#include "picongpu/plugins/IsaacPlugin.hpp"
+#include "picongpu/plugins/PhaseSpace/PhaseSpace.hpp"
+#include "picongpu/plugins/PluginRegistry.hpp"
+#include "picongpu/plugins/PngPlugin.hpp"
 #include "picongpu/plugins/SumCurrents.hpp"
+#include "picongpu/plugins/binning/BinningDispatcher.hpp"
+#include "picongpu/plugins/makroParticleCounter/PerSuperCell.hpp"
 #include "picongpu/plugins/multi/Master.hpp"
-#include "picongpu/plugins/output/images/PngCreator.hpp"
-#include "picongpu/plugins/output/images/Visualisation.hpp"
+#include "picongpu/plugins/openPMD/openPMDWriter.hpp"
+#include "picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp"
+#include "picongpu/plugins/radiation/Radiation.hpp"
+#include "picongpu/plugins/radiation/VectorTypes.hpp"
+#include "picongpu/plugins/shadowgraphy/Shadowgraphy.hpp"
 #include "picongpu/plugins/transitionRadiation/TransitionRadiation.hpp"
 
 #include <pmacc/assert.hpp>
-/* That's an abstract plugin for image output with the possibility
- * to store the image as png file or send it via a sockets to a server.
- *
- * \todo rename PngPlugin to ImagePlugin or similar
- */
-#include "picongpu/plugins/PngPlugin.hpp"
-
-#if(SIMDIM == DIM3 && PIC_ENABLE_FFTW3 == 1 && ENABLE_OPENPMD == 1)
-#    include "picongpu/plugins/shadowgraphy/Shadowgraphy.hpp"
-#endif
-
-#if(ENABLE_OPENPMD == 1)
-#    include "picongpu/plugins/PhaseSpace/PhaseSpace.hpp"
-#    include "picongpu/plugins/binning/BinningDispatcher.hpp"
-#    include "picongpu/plugins/openPMD/openPMDWriter.hpp"
-#    include "picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp"
-#endif
-
-#include "picongpu/plugins/ChargeConservation.hpp"
-#if(ENABLE_OPENPMD == 1)
-#    include "picongpu/plugins/makroParticleCounter/PerSuperCell.hpp"
-#endif
-
-#if(ENABLE_ISAAC == 1) && (SIMDIM == DIM3)
-#    include "picongpu/plugins/IsaacPlugin.hpp"
-#endif
-
-#if ENABLE_OPENPMD
-#    include "picongpu/plugins/radiation/Radiation.hpp"
-#    include "picongpu/plugins/radiation/VectorTypes.hpp"
-#endif
-
-#include "picongpu/particles/traits/SpeciesEligibleForSolver.hpp"
-#include "picongpu/plugins/Checkpoint.hpp"
-#include "picongpu/plugins/ILightweightPlugin.hpp"
-#include "picongpu/plugins/ISimulationPlugin.hpp"
-
 #include <pmacc/mappings/kernel/MappingDescription.hpp>
 #include <pmacc/meta/AllCombinations.hpp>
 
-#include <list>
+#include <algorithm>
```

Comment:
```suggestion
#include <vector>
```
In your current version, `algorithm` isn't used. `vector` is.